src/util/models.py

```diff
@@ -162,6 +163,10 @@ def remove_model(cr, model, drop_table=True, ignore_m2m=()):
 
     _remove_import_export_paths(cr, model)
 
+    if remove_inherits:
+        for inh in for_each_inherit(cr, model):
```

Comment:
We could do this always and output a warning (with the name of the flag to set), advantages:
* In CI developers will know what to do
* In older versions, or client's upgrades, we will see in the log how widespread this issue is.